### PR TITLE
Update route management API endpoints

### PR DESCRIPTION
**Pull request by Pixcell**
- Does the iofog.org or README.md documentation need to change?
  - [x] Yes https://github.com/eclipse-iofog/iofog.org/pull/125
  - [ ] No

### CODE REVIEW

File: src/data/migrations/20200506004924-add-route-name.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.addColumn('Routings', 'name', {
+      type: Sequelize.TEXT,
+      allowNull: false,
+      defaultValue: 'route'
```
Comment (baghbidi) [inline src/data/migrations/20200506004924-add-route-name.js:8]: unique index shouldn't have default value imo

File: src/helpers/error-messages.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ module.exports = {
   ALREADY_EXISTS: 'Model already exists',
   INVALID_CATALOG_ITEM_ID: 'Invalid catalog item id \'{}\'',
   INVALID_FLOW_ID: 'Invalid flow id \'{}\'',
+  INVALID_ROUTING_ID: 'Invalid route name \'{}\'',
```
Comment (baghbidi) [inline src/helpers/error-messages.js]: `INVALID_ROUTING_NAME`?

File: src/services/routing-service.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  // create new route
+  const createRouteData = {
+    ...routeData,
+    sourceMicroserviceUuid: sourceMicroservice.uuid,
+    destMicroserviceUuid: destMicroservice.uuid
+  }
+
+  const newRoute = await RoutingManager.create(createRouteData, transaction)
+  await _switchOnUpdateFlagsForMicroservicesInRoute(sourceMicroservice, destMicroservice, transaction)
+  console.log({ newRoute })
```
Comment (baghbidi) [inline src/services/routing-service.js]: delete this

**Review by baghbidi — CHANGES_REQUESTED** [review]

File: src/services/routing-service.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  if (sourceMicroservice.flowId !== destMicroservice.flowId) {
+    throw new Errors.ValidationError('microservices on different flows')
+  }
+
+  const route = await RoutingManager.findOne({
+    sourceMicroserviceUuid: sourceMicroservice.uuid,
+    destMicroserviceUuid: destMicroservice.uuid
+  }, transaction)
+  if (route) {
+    throw new Errors.ValidationError('route already exists')
```
Comment (baghbidi) [inline src/services/routing-service.js]: `Errors.DuplicatePropertyError`

File: src/data/migrations/20200506004924-add-route-name.js (hunk 1 above)
Comment (Pixcell) [reply src/data/migrations/20200506004924-add-route-name.js:8]: It does not let me do `allowNull: false` without a default value :S But I agree with you

**Review by baghbidi — APPROVED** [review]

**Pull request merged**